iandavis/answers.js

```diff
@@ -0,0 +1,206 @@
+
+//m
+++x == x++
+//true if x equals any number, including Infinity, though not NaN, because x is incremented left of the equality operator, then remains the same right of the equality operator. Also true for strings that are quoted numbers, which are converted to numbers. Also true if x is a boolean value.
+
+//n
+"1" + x == 1 + x
+// only true if x is a string.  if x is anything else it is converted to a string right of equality and the left of equality never evaluates to a string
+
+
+//o
+"0" + 1 == 1
+//true because the string "01" is converted to 1 by the equality operator.
+
+
+//p
+(typeof (x + 1)) === (typeof x)
+//true is x is any string or number, including NaN and Infinity. For a string x it is concatenated with 1, both side of the strict equality being strings. False if x is boolean because booleans are reduced to a number when added to a number. undefined is false because undefined + 1 is NaN, a number, while typeof undefined is "undefined".
+
+
+//q
+(x * 1 == x) || ((typeof x) != "number")
+//true except when x is NaN because does not equal itself (left of ||) and is a number.
+//left of || is true in all but two other cases. when x is a string that isn't a quoted number (with that it is converted to a number with the * operator) it evaluates to NaN which is not equal to itself. In this case, right of ||, and evaluates to true. The other case is when x is undefined. left of || evaluates to NaN, right of || evaluates to true, typeof undefined being the string "undefined".
+
+//r
+(x = (typeof (x + (typeof x)))) == x
+// always true, including if x is NaN, because typeof always outputs a string
+
+
+var testCases = ["string", "1", true, false, Infinity, 20, undefined, NaN];
+
+for (var x = 0, length = testCases.length; x < length; x++) {
+	var result = (testCases[x] = (typeof (testCases[x] + (typeof testCases[x])))) == testCases[x];
+	console.log(testCases[x] + " is " + result)
+};
+
+
+/* SECTION 2*/
+
+//a) Write an expression for the mean (i.e. average) of x, y, and z.
+(x + y + z) / 3;
+
+
+//b) Write a series of expressions to adjust each of x, y, and z halfway toward their mean. That is, reset the value of each variable to something based on its previous value.
+
+var x = 200, y = 500, z = 300;
+var mean = (x + y + z) / 3;
+
+x < average ? (x += (average - x) / 2) : (x -= (x - average) / 2);
+y < average ? (y += (average - y) / 2) : (y -= (y - average) / 2);
+z < average ? (z += (average - z) / 2) : (z -= (z - average) / 2);
+
+
+/* SECTION 3*/
+
+//a) Write an expression for the rectangle's area.
+var width = r - l, height = t - b;
+var area = width * height;
+
+//b) Write an expression which is true if the rectangle is taller than it is wide, and false otherwise.
+
+height > width > ? true : false
+(height > width)|| false
+
```

Comment:
Notice that A+=B is the same as A-= -B.  Therefore `(x+=(average-x))` is the same as `(x-=(x-average))`.  So the conditional isn't actually needed!  You did use it correctly, though, so points there!